src/commands/types.py

```diff
@@ -19,3 +19,45 @@ class CommandDescriptor:
     action: str
     params: Mapping[str, Any]
     icon: str | None = None
+
+
+@runtime_checkable
+class HandlerProtocol(Protocol):
+    """Protocol for command handlers."""
+
+    def run(self, caller: Any, **kwargs: Any) -> None:
```

Comment:
Caller shouldn't be Any type. For Evennia, we know that a caller must be one of three types of instances: An AccountDB instance (which for our purposes, is usually turned into an Account Typeclass), a Session object, or an ObjectDB instance (usually turned into a Character typeclass instance, but not always). We may want to create a type to reflect that elsewhere, possibly in the evennia extensions app.